src/dids/OracleResolveDriver.ts

```diff
@@ -0,0 +1,134 @@
+            var desiredVal = "";
+            while(initialPos < finalPos){
+                desiredVal += fileContents[initialPos];
+                initialPos++;
+            }
+            return desiredVal;
+        }else{// option 4
+            var initialPos = fileContents.indexOf("CHAINCODE") + 9;
+            var finalPos = fileContents.length;
+            var desiredVal = "";
+            while(initialPos < finalPos){
+                desiredVal += fileContents[initialPos];
+                initialPos++;
+            }
+            return desiredVal;
+        }
+    }
+
+    /**
+     * Method to query user's config file, which contains ledger specific information
+     * @param option number of which piece of information you need from config file
+     * @summary 
+     * Option 1: CHAINCODE string
+     * Option 2: NETWORK string
+     * Option 3: CHANNEL string
+     * Option 4: ENCODEDCRED string
+     * @returns string of ledger endpoints to hit with REST API calls
+     */
+    public configQuery(option:number, configFileStr:string): string{
+        //check that option is in range 1 to 4
+        if(option < 1 || option > 4 || configFileStr.length == 0){
+            throw new Error("Invalid params passed to configQuery function");
+        }
+
+        //test with mock output
+        if(configFileStr == "TEST"){//give mock output
+            const mockData: string = fs.readFileSync(process.cwd() + "/transcripts/testConfig.txt",'utf-8');
+            return OracleResolveDriver.parseConfig(option, mockData);
+        }
+
+        //actual personal config file query
+        const fileData: string = fs.readFileSync(process.cwd() + configFileStr, 'utf-8');
+        return OracleResolveDriver.parseConfig(option, fileData);
+    }
+
+    /**
+     * Method to use queries config file information and make REST call to associated ledger to get DIDDoc for associated DID
+     * @param did DID to be resolved as string type
+     * @returns Promise of DIDDoc resolution or error type defined in Open-Wallet Credo library
+     */
+    async didResolve(did: string, queryDirectory: string): Promise<any>{
+        if(did.length <= 0 || queryDirectory.length <= 0){
+                if(did.length <= 0){
+                    return -1;
+                }else{
+                    return -1;
+                }
+        }
+        //get did data with passed did and config ledger chaincode like OracleLedgerService line 131
+        var chaincodeStr = this.configQuery(1,queryDirectory);
+        var didDoc = JSON.stringify({
+            chaincode: chaincodeStr,
+            args: ["GetDidDocumentById", did],
+            sync: true,
+        });
+        //hit axios post with network info like OracleLedgerSerivce line 137 and line 148
+        var networkStr = this.configQuery(2,queryDirectory);
+        var channelStr = this.configQuery(3, queryDirectory);
+        var credStr = this.configQuery(4, queryDirectory);
```

Comment:
You could use array destructuring to get cleaner code. If you want.
```typescript
const [chaincodeStr, networkStr, channelStr, credStr] = await Promise.all([
      this.configQuery(1, queryDirectory),
      this.configQuery(2, queryDirectory),
      this.configQuery(3, queryDirectory),
      this.configQuery(4, queryDirectory)
    ]);

const data...
```

Reply:
configQuery is a synchronous function.  However, maybe change the variables to be const instead of var to ensure no manipulation before use?

Reply:
True, sorry I glassed over the readFile**Sync**(). Yes const is better. You can still do destructuring though for cleaner code.

It is not good practice to use var at all actually (only very rare scenarios). `let` and `const` are better.